CMSgov/mint-app
# [EASI-4308] [EASI-4310] EASI-4311] FE work for adding New Model Plan Notifications

# EASI-4308, EASI-4310, EASI-4311


## Changes and Description

- Updated Setttings page with the new additional notifications
- Updated `IndividualNotification` component wtih `isNewModelPlan` type guard
- Updated Type Guard utils file (src/views/Notifications/Home/_components/_utils.tsx) 
    - DRY code
    - added `isNewModelPlan`
    - Updated translations to use new enum values
- Add `newModelPlan` to graphQL
- Updated Notification translation file
    - to use the enum values
    - to have copy updates, according to figma
    - to have unsubscription alert banner copy
- Minor updates: updated `cypress/support/model-plan-by-name.js` to make it easier to read
- Updated tests and snaps
- Updated Notification Cypress tests
- Miscellaneous code clean up
- 


## How to test this change

1. fresh seed the app
2. Navigate to Notification settings page
3. verify that notifications for `when a new Model Plan is created` are not checked
4. check email & in-app notification && save
5. Ensure data is saved and persists
6. Create a new model plan
7. Check notification to see that the new model plan creation notification is there
8. Navigate to `http://localhost:3005/notifications/settings?unsubscribe_email=NEW_MODEL_PLAN`
9. Verify new model plan email notification is unchecked and that banner matches figma
10. Navigate again to `http://localhost:3005/notifications/settings?unsubscribe_email=NEW_MODEL_PLAN` 
11. Verify that error message shows up and matches figma
12. Check email notification again and save
13. Ensure data saves and persists
14. try breaking things



## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
